server/src/main/scala/drt/chroma/ArrivalsDiffingStage.scala

```diff
@@ -53,36 +52,32 @@ final class ArrivalsDiffingStage(initialKnownArrivals: Seq[Arrival]) extends Gra
       maybeResponseToPush = None
     }
 
-    def processFeedResponse(ArrivalsFeedResponse: ArrivalsFeedResponse): Option[ArrivalsFeedResponse] = {
-      ArrivalsFeedResponse match {
-        case afs@ArrivalsFeedSuccess(latestArrivals, _) =>
-          val newUpdates = diff(knownArrivals, latestArrivals.flights)
-          log.info(s"Got ${newUpdates.length} new arrival updates")
-          knownArrivals = latestArrivals.flights
-          Option(afs.copy(arrivals = latestArrivals.copy(flights = newUpdates)))
-        case aff@ArrivalsFeedFailure(_, _) =>
-          log.info("Passing ArrivalsFeedFailure through. Nothing to diff. No updates to knownArrivals")
-          Option(aff)
-        case unexpected =>
-          log.error(s"Unexpected ArrivalsFeedResponse: ${unexpected.getClass}")
-          Option.empty[ArrivalsFeedResponse]
-      }
+    def processFeedResponse(ArrivalsFeedResponse: ArrivalsFeedResponse): Option[ArrivalsFeedResponse] = ArrivalsFeedResponse match {
+      case afs@ArrivalsFeedSuccess(latestArrivals, _) =>
+        val incomingArrivals = latestArrivals.flights.map(a => (ArrivalKey(a), a))
+        val newUpdates = diff(knownArrivals, incomingArrivals)
```

Comment:
potentially worth adding type annotations to the above two lines for clarity.

Reply:
I've added annotations and renamed `diff` to `filterArrivalsWithUpdates`